### ISSUE TITLE
how to keep sidebar open when click sub menu[v2.4.2]

### ISSUE THREAD

**Issue by huangjianke** [almasaeed2010/AdminLTE#1806]
Hello, Thanks for your greate job.
I use adminLTE in my Flask project, but when I click sub menu, all menu will be refresh and closed, I have already spend too many time for this, please help me

**Comment by huangjianke** [comment]
I have add a sidebar in  base_site.html, all views was extends the base_site.html,So, how can I keep the sidebar menu open when click sub menu, If anyone can help, I would be very wonderful, thanks

![image](https://user-images.githubusercontent.com/8692455/35680232-b6a3b592-0794-11e8-82d7-3a59c444c607.png)
base_site.html

**Comment by huangjianke** [comment]
@almasaeed2010 @REJack

**Comment by REJack** [comment]
Please provide your code from sidebar.html too or a complete rendered version of the whole template.

**Comment by huangjianke** [comment]
[code.zip](https://github.com/almasaeed2010/AdminLTE/files/1687757/code.zip)
@REJack here is the code of sidebar.html, base_site.html, thanks

**Comment by huangjianke** [comment]
![demo](https://user-images.githubusercontent.com/8692455/35712163-a5e49516-07fb-11e8-8f86-2d0c4e4c3cd0.gif)

**Comment by REJack** [comment]
Oh ok you need the active class on your sidebar entries, 
you can do this either through the Flask's templating engine Jinja2 they have a nice [Highlighting Active Menu Items](http://jinja.pocoo.org/docs/2.10/tricks/#highlighting-active-menu-items) tutorial/example or you add a simple javascript with this lines:
```
/** add active class and stay opened when selected */
var url = window.location;

// for sidebar menu entirely but not cover treeview
$('ul.sidebar-menu a').filter(function() {
	 return this.href == url;
}).parent().addClass('active');

// for treeview
$('ul.treeview-menu a').filter(function() {
	 return this.href == url;
}).parentsUntil(".sidebar-menu > .treeview-menu").addClass('active');
```

**Comment by huangjianke** [comment]
@REJack Good job，Thanks

**Comment by ckoleysmartsol** [comment]
Thanks @REJack .. it works... Thank you so much

**Comment by coderMmw** [comment]
> Oh ok you need the active class on your sidebar entries,
> you can do this either through the Flask's templating engine Jinja2 they have a nice [Highlighting Active Menu Items](http://jinja.pocoo.org/docs/2.10/tricks/#highlighting-active-menu-items) tutorial/example or you add a simple javascript with this lines:
> 
> ```
> /** add active class and stay opened when selected */
> var url = window.location;
> 
> // for sidebar menu entirely but not cover treeview
> $('ul.sidebar-menu a').filter(function() {
> 	 return this.href == url;
> }).parent().addClass('active');
> 
> // for treeview
> $('ul.treeview-menu a').filter(function() {
> 	 return this.href == url;
> }).parentsUntil(".sidebar-menu > .treeview-menu").addClass('active');
> ```

hello! I have the same problem, do as you say, but it still has no effect.
here is my code
`<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<title>AdminLTE 3 | Dashboard</title>
<!-- Tell the browser to be responsive to screen width -->
<meta name="viewport" content="width=device-width, initial-scale=1">
<!-- Font Awesome -->
<link rel="stylesheet" href="plugins/fontawesome-free/css/all.min.css">
<!-- Ionicons -->
<link rel="stylesheet"
	href="https://code.ionicframework.com/ionicons/2.0.1/css/ionicons.min.css">
<!-- Tempusdominus Bbootstrap 4 -->
<link rel="stylesheet"
	href="plugins/tempusdominus-bootstrap-4/css/tempusdominus-bootstrap-4.min.css">

<!-- iCheck -->
<link rel="stylesheet"
	href="plugins/icheck-bootstrap/icheck-bootstrap.min.css">
<!-- JQVMap -->
<link rel="stylesheet" href="plugins/jqvmap/jqvmap.min.css">
<!-- Theme style -->
<link rel="stylesheet" href="dist/css/adminlte.min.css">
<!-- overlayScrollbars -->
<link rel="stylesheet"
	href="plugins/overlayScrollbars/css/OverlayScrollbars.min.css">
<!-- Daterange picker -->
<link rel="stylesheet"
	href="plugins/daterangepicker/daterangepicker.css">
<!-- summernote -->
<link rel="stylesheet" href="plugins/summernote/summernote-bs4.css">


<link rel="stylesheet" type="text/css" href="plugins/jsgrid/jsgrid.css" />
<link rel="stylesheet" type="text/css"
	href="plugins/jsgrid/jsgrid-theme.min.css" />

<!-- Google Font: Source Sans Pro -->
<link
	href="https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,400,400i,700"
	rel="stylesheet">
<link rel="stylesheet" href="http://code.jquery.com/ui/1.11.2/themes/cupertino/jquery-ui.css">
<style>
.ui-widget *, .ui-widget input, .ui-widget select, .ui-widget button {
	font-family: 'Helvetica Neue Light', 'Open Sans', Helvetica;
	font-size: 14px;
	font-weight: 300 !important;
}

.details-form-field input, .details-form-field select {
	width: 250px;
	float: right;
}

.details-form-field {
	margin: 30px 0;
}

.details-form-field:first-child {
	margin-top: 10px;
}

.details-form-field:last-child {
	margin-bottom: 10px;
}

.details-form-field button {
	display: block;
	width: 100px;
	margin: 0 auto;
}

input.error, select.error {
	border: 1px solid #ff9999;
	background: #ffeeee;
}

label.error {
	float: right;
	margin-left: 100px;
	font-size: .8em;
	color: #ff6666;
}
</style>
</head>
<body class="hold-transition sidebar-mini layout-fixed">
	<div class="wrapper">

		<!-- Navbar -->

		<!-- SEARCH FORM -->

		<!-- Right navbar links -->
		<!-- /.navbar -->

		<!-- Main Sidebar Container -->
		<aside class="main-sidebar sidebar-dark-primary elevation-4">
			<!-- Brand Logo -->
			<a href="index3.html" class="brand-link"> <img
				src="dist/img/AdminLTELogo.png" alt="AdminLTE Logo"
				class="brand-image img-circle elevation-3" style="opacity: .8">
				<span class="brand-text font-weight-light">AdminLTE 3</span>
			</a>
			
			<!-- Sidebar -->
			<div class="sidebar">
				<!-- Sidebar user panel (optional) -->



				<!-- Sidebar Menu -->
				<nav class="mt-2">
					<!--nav-child-indent 父子阶梯效果  -->
					<ul
						class="nav  nav-treeview nav-pills nav-sidebar nav-child-indent flex-column"
						data-widget="treeview" role="menu" data-accordion="false">
						<!-- Add icons to the links using the .nav-icon class
               with font-awesome or any other icon font library -->


						<li class="nav-item has-treeview"><a href="#"
							class="nav-link"> <i class="nav-icon fas fa-table"></i>
								<p>
									编排管理 <i class="fas fa-angle-left right"></i>
								</p>
						</a>
							<ul class="nav nav-treeview ">
								<li class="nav-item has-treeview"><a href=""
									class="nav-link"> <i class="far fa-circle nav-icon"></i>
										<p>内容编排</p> <i class="fas fa-angle-left right"></i>

								</a>
									<ul class="nav nav-treeview">
										<span></span>
										<li class="nav-item"><a href="./new_index.html"
											class="nav-link"> <i class="far fa-circle nav-icon"></i>
												<p>xxx栏目</p>
										</a> </a></li>
									</ul></li>
								<li class="nav-item"><a href="" class="nav-link"> <i
										class="far fa-circle nav-icon"></i>
										<p>栏目管理</p> <i class="fas fa-angle-left right"></i>
								</a></li>
							</ul></li>





















						<li class="nav-item has-treeview"><a href="#"
							class="nav-link"> <i class="nav-icon fas fa-table"></i>
								<p>
									Tables <i class="fas fa-angle-left right"></i>
								</p>
						</a>
							<ul class="nav nav-treeview ">
								<li class="nav-item has-treeview"><a
									href="pages/tables/simple.html" class="nav-link"> <i
										class="far fa-circle nav-icon"></i>
										<p>Simple Tables</p>
								</a></li>
								<li class="nav-item"><a href="pages/tables/data.html"
									class="nav-link"> <i class="far fa-circle nav-icon"></i>
										<p>DataTables</p>
								</a></li>
								<li class="nav-item"><a href="pages/tables/jsgrid.html"
									class="nav-link"> <i class="far fa-circle nav-icon"></i>
										<p>jsGrid</p>
								</a></li>
							</ul></li>

					</ul>
				</nav>
				<!-- /.sidebar-menu -->
			</div>
			<!-- /.sidebar -->
		</aside>

		<!-- Content Wrapper. Contains page content -->
		<div class="content-wrapper">
			<!-- Content Header (Page header) -->
			<div class="content-header"></div>
			<!-- /.content-header -->

			<!-- Content Wrapper. Contains page content -->
			<!-- Content Header (Page header) -->
			<section class="content-header">
				<div class="container-fluid">
					<div class="row mb-2">
						<div class="col-sm-6">
							<h1>jsGrid</h1>
						</div>
						<div class="col-sm-6">
							<ol class="breadcrumb float-sm-right">
								<li class="breadcrumb-item"><a href="#">Home</a></li>
								<li class="breadcrumb-item active">jsGrid1</li>
							</ol>
						</div>
					</div>
				</div>
				<!-- /.container-fluid -->
			</section>
			<!-- Main content -->
			<section class="content">
				<div class="card">
					<div class="card-header">
						<h3 class="card-title"></h3>
						<div class="demo"></div>
					</div>
					<!-- /.card-header -->
					<div class="card-body"></div>
					<div id="jsGrid"></div>
					<div id="detailsDialog">
						<form id="detailsForm">
							<div class="details-form-field">
								<label for="name">Name:</label> <input id="name" name="name"
									type="text" />
							</div>
							<div class="details-form-field">
								<label for="age">Age:</label> <input id="age" name="age"
									type="number" />
							</div>
							<div class="details-form-field">
								<label for="address">Address:</label> <input id="address"
									name="address" type="text" />
							</div>
							<div class="details-form-field">
								<label for="country">Country:</label> <select id="country"
									name="country">
									<option value="">(Select)</option>
									<option value="1">United States</option>
									<option value="2">Canada</option>
									<option value="3">United Kingdom</option>
									<option value="4">France</option>
									<option value="5">Brazil</option>
									<option value="6">China</option>
									<option value="7">Russia</option>
								</select>
							</div>
							<div class="details-form-field">
								<label for="married">Is Married</label> <input id="married"
									name="married" type="checkbox" />
							</div>
							<div class="details-form-field">
								<button type="submit" id="save">Save</button>
							</div>
						</form>
					</div>


					<!-- /.card-body -->
				</div>
				<!-- /.card -->
			</section>
			
			<!-- /.content -->
			<footer class="main-footer">
				<strong>Copyright &copy; 2014-2019 <a
					href="http://adminlte.io">AdminLTE.io</a>.
				</strong> All rights reserved.
				<div class="float-right d-none d-sm-inline-block">
					<b>Version</b> 3.0.0-rc.1
				</div>
			</footer>

			<!-- Control Sidebar -->
			<!--  <aside class="control-sidebar control-sidebar-dark">
    Control sidebar content goes here
  </aside> -->
			<!-- /.control-sidebar -->
		</div>
		<!-- ./wrapper -->

		<!-- jQuery -->
		<script src="plugins/jquery/jquery.min.js"></script>
		<!-- jQuery UI 1.11.4 -->
		<script src="plugins/jquery-ui/jquery-ui.min.js"></script>
		<!-- Resolve conflict in jQuery UI tooltip with Bootstrap tooltip -->
		<!-- <script type="text/javascript">
  $.widget.bridge('uibutton', $.ui.button)
  $(function () {
    $("#jsGrid1").jsGrid({
        height: "100%",
        width: "100%",
        sorting: true,
        paging: true,
 		data: db.clients,
        filtering: true,
        inserting: true,
        autoload: true,
        fields: [
            { name: "Name", type: "text", width: 150 },
            { name: "Age", type: "number", width: 50 },
            { name: "Address", type: "text", width: 200 },
            { name: "Country", type: "select", items: db.countries, valueField: "Id", textField: "Name" },
            { name: "Married", type: "checkbox", title: "Is Married" },
            {type:"control"}
        ]
    });
  });

</script> -->
		<script>

  
  /*$('#demo').daterangepicker({
    "startDate": "09/05/2019",
    "endDate": "09/11/2019"
}, function(start, end, label) {
  console.log('New date range selected: ' + start.format('YYYY-MM-DD') + ' to ' + end.format('YYYY-MM-DD') + ' (predefined range: ' + label + ')');
});*/


/** add active class and stay opened when selected */
var url = window.location;

// for sidebar menu entirely but not cover treeview
$('ul.nav').filter(function() {
	 return this.href == url;
}).parent().addClass('active');

// for treeview
$('ul.nav a').filter(function() {
	 return this.href == url;
}).parentsUntil(".nav > .nav-item").addClass('active');





   $(function() {

            $("#jsGrid").jsGrid({
                //如果设置成100%则还出现高度没了的现象
                height: "auto",
                width: "100%",
                editing: true,
                autoload: true,
                paging: true,
                deleteConfirm: function(item) {
                    return "The client \"" + item.Name + "\" will be removed. Are you sure?";
                },
                rowClick: function(args) {
                    showDetailsDialog("Edit", args.item);
                },
                controller: db,
                fields: [
                    { name: "Name", type: "text", width: 150 },
                    { name: "Age", type: "number", width: 50 },
                    { name: "Address", type: "text", width: 200 },
                    { name: "Country", type: "select", items: db.countries, valueField: "Id", textField: "Name" },
                    { name: "Married", type: "checkbox", title: "Is Married", sorting: false },
                    {
                        type: "control",
                        modeSwitchButton: false,
                        editButton: false,
                        headerTemplate: function() {
                            return $("<button>").attr("type", "button").text("Add")
                                .on("click", function () {
                                    showDetailsDialog("Add", {});
                                });
                        }
                    }
                ]
            });

            $("#detailsDialog").dialog({
                autoOpen: false,
                width: 400,
                close: function() {
                    $("#detailsForm").validate().resetForm();
                    $("#detailsForm").find(".error").removeClass("error");
                }
            });

            $("#detailsForm").validate({
                rules: {
                    name: "required",
                    age: { required: true, range: [18, 150] },
                    address: { required: true, minlength: 10 },
                    country: "required"
                },
                messages: {
                    name: "Please enter name",
                    age: "Please enter valid age",
                    address: "Please enter address (more than 10 chars)",
                    country: "Please select country"
                },
                submitHandler: function() {
                    formSubmitHandler();
                }
            });

            var formSubmitHandler = $.noop;

            var showDetailsDialog = function(dialogType, client) {
                $("#name").val(client.Name);
                $("#age").val(client.Age);
                $("#address").val(client.Address);
                $("#country").val(client.Country);
                $("#married").prop("checked", client.Married);

                formSubmitHandler = function() {
                    saveClient(client, dialogType === "Add");
                };

                $("#detailsDialog").dialog("option", "title", dialogType + " Client")
                    .dialog("open");
            };

            var saveClient = function(client, isNew) {
                $.extend(client, {
                    Name: $("#name").val(),
                    Age: parseInt($("#age").val(), 10),
                    Address: $("#address").val(),
                    Country: parseInt($("#country").val(), 10),
                    Married: $("#married").is(":checked")
                });

                $("#jsGrid").jsGrid(isNew ? "insertItem" : "updateItem", client);

                $("#detailsDialog").dialog("close");
            };

        });
</script>
		<!-- Bootstrap 4 -->
		<script src="plugins/bootstrap/js/bootstrap.bundle.min.js"></script>
		<!-- ChartJS -->
		<script src="plugins/chart.js/Chart.min.js"></script>
		<!-- Sparkline -->
		<script src="plugins/sparklines/sparkline.js"></script>
		<!-- JQVMap -->
		<script src="plugins/jqvmap/jquery.vmap.min.js"></script>
		<script src="plugins/jqvmap/maps/jquery.vmap.world.js"></script>
		<!-- jQuery Knob Chart -->
		<script src="plugins/jquery-knob/jquery.knob.min.js"></script>

		<script src="http://code.jquery.com/jquery-1.10.2.js"></script>
		<script src="http://code.jquery.com/ui/1.11.2/jquery-ui.js"></script>
		<script
			src="http://ajax.aspnetcdn.com/ajax/jquery.validate/1.9/jquery.validate.min.js"></script>

		<!-- daterangepicker -->
		<script src="plugins/moment/moment.min.js"></script>
		<script src="plugins/daterangepicker/daterangepicker.js"></script>
		<!-- Tempusdominus Bootstrap 4 -->
		<script
			src="plugins/tempusdominus-bootstrap-4/js/tempusdominus-bootstrap-4.min.js"></script>
		<!-- Summernote -->
		<script src="plugins/summernote/summernote-bs4.min.js"></script>
		<!-- overlayScrollbars -->
		<script
			src="plugins/overlayScrollbars/js/jquery.overlayScrollbars.min.js"></script>
		<!-- AdminLTE App -->
		<script src="dist/js/adminlte.js"></script>
		<!-- AdminLTE dashboard demo (This is only for demo purposes) -->
		<script src="plugins/jsgrid/demos/db.js"></script>
		<script src="plugins/jsgrid/jsgrid.min.js"></script>
		<!-- <script src="dist/js/pages/dashboard.js"></script> -->
		<!-- AdminLTE for demo purposes -->
		<script src="dist/js/demo.js"></script>
</body>
</html>
`

**Comment by DariusDarvish** [comment]
Hi, I was wondering how we can keep the parent directory open and highlight the current page we are on.
 Something like the link below:
https://adminlte.io/themes/AdminLTE/pages/charts/flot.html

Thank You,
Darius
